src/JustSaying/Messaging/Middleware/DefaultSqsMiddleware.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Amazon.SQS.Model;
+using JustSaying.Messaging.Channels;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Middleware
+{
+    public class DefaultSqsMiddleware : MiddlewareBase<GetMessagesContext, IList<Amazon.SQS.Model.Message>>
+    {
+        private readonly ILogger<DefaultSqsMiddleware> _logger;
+
+        public DefaultSqsMiddleware(ILogger<DefaultSqsMiddleware> logger) : base(null)
+        {
+            _logger = logger;
+        }
+
+        protected override async Task<IList<Message>> RunInnerAsync(GetMessagesContext context, Func<Task<IList<Message>>> func)
+        {
+            try
+            {
+                var results = await func().ConfigureAwait(false);
+
+                _logger.LogTrace(
+                    "Polled for messages on queue '{QueueName}' in region '{Region}', and received {MessageCount} messages.",
+                    context.QueueName,
+                    context.RegionName,
+                    context.Count);
+
+                return results;
+            }
+            catch (InvalidOperationException ex)
```

Comment:
How does this exception type indicate that the count of messages to handle failed?

Reply:
This exception is no longer thrown by the new message receive buffer, so we'll remove this exception handler.